### ISSUE TITLE
1 click output error/warning autofix

### ISSUE THREAD

**Issue by Directory** [intellij-rust/intellij-rust#6827]
this is a feature request then an issue but it would be super awsome.
run this code
```rust
fn main() {
    let mut s = 55;
    println!("{}", s);
    let s = 565555;
    println!("{}", s);
}
```
it will raise this warning but still compile and execute
```rust
C:/Redacted/.cargo/bin/cargo.exe run --color=always --package redacted --bin redacted
warning: variable does not need to be mutable
 --> src\main.rs:2:9
  |
2 |     let mut s = 55;
  |         ----^
  |         |
  |         help: remove this `mut`
  |
  = note: `#[warn(unused_mut)]` on by default

warning: 1 warning emitted

    Finished dev [unoptimized + debuginfo] target(s) in 0.02s
     Running `target\debug\redacted.exe`
55
565555

Process finished with exit code 0
```
now run this code
```rust
fn main() {
    println!("{}", x());
}
fn x() -> i32 {
    1 + 1;
}

```
it will not execute but give a compile error in build output
```rust
error[E0308]: mismatched types
 --> src\main.rs:5:11
  |
5 | fn x() -> i32 {
  |    -      ^^^ expected `i32`, found `()`
  |    |
  |    implicitly returns `()` as its body has no tail or `return` expression
6 |     1 + 1;
  |          - help: consider removing this semicolon
```
these examples show rust's excellent compiler adding auto suggestions for fixes. rather then inconveniencing the programmer by making them do the pain staking task of moving their fingers slightly to fix code wouldn't it be awesome if you could just click the `help` suggestion in the output and for the code to auto fix itself? it would be a very nice quality of life improvement if you could just click away your problems like a google doc suggestion. after all the compiler knows best and the exact change needed is right there, why type it out yourself.

i dont know how this would be implemented but i imagine the build output would need some serious regex to identify offered `help` fixes, get the character range and module after the `--> ` arrow, and the compiler suggested line to replace. maybe the compiler has an api that can get this data directly without the need for stdout parsing idk. im mostly unsure of how the plugin can read mouse input clicking on the build output help suggestion if thats even possible. it would also be cool if the output line could have some sort of gui button or overlay on the help suggestion to make it obvious its a clickable element. or if it would utilize intellij's gui git style code change comparison thing you see when commiting. idk how intellij plugins work so im making a lot of concept ideas.

**Comment by Undin** [comment]
@Directory thanks for the idea!
But I'm not sure that the suggested feature should be implemented in this way. My main concern that build output may be outdated and it may be quite tricky to handle it. Also, it's quite hard to extract necessary "help" info from build output properly.
Fortunately, the plugin already has a similar feature called `External Linter`.
There are two different ways to use it:
- run on demand. You can do it via `Cargo` tool window or via `Analyze | Run Inspection by Name` action and type `External linter` in the corresponding dialog

https://user-images.githubusercontent.com/2539310/107738731-56edcd80-6d18-11eb-83c1-5bcd14c51fe1.mov

- run on the fly. Just turn on `Run external linter to analyze code on the fly` option in `Preferences | Languages & Frameworks | Rust | Cargo` settings. With this option enabled the plugin runs `cargo check` on each change in the editor (yep, it's resource-consuming so may not work perfectly for large codebase) and annotate your code.

https://user-images.githubusercontent.com/2539310/107738759-640abc80-6d18-11eb-901b-94a91f3b6cae.mov
